src/core/jpg.js

```diff
@@ -148,8 +149,9 @@ var JpegImage = (function JpegImageClosure() {
         var nextByte = data[offset++];
         if (nextByte) {
           if (nextByte === 0xdc && parseDNLMarker) {
-            offset += 2; // Skip data length.
-            const scanLines = (data[offset++] << 8) | data[offset++];
+            offset += 2; // Skip marker length.
+
+            const scanLines = readUint16(data, offset);
```

Comment:
Is this equivalent? `readUint16` does `(data[offset] << 8) | data[offset + 1]` whereas the previous code did `(data[offset++] << 8) | data[offset++]`, so the offset is incremented for the bitwise shift too.

Reply:
You're correct, in order for the new code to be completely identical to the old one there should be a `offset += 2;` line afterwards.
In this *particular* case it really doesn't matter since there's an Error thrown below, but generally this probably isn't great if this code is ever re-factored; I'll just fix this to avoid any future issues.